test/helper/e2e2/kube/kube.go

```diff
@@ -0,0 +1,178 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/api/v1"
+	akov2next "github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/nextapi/v1"
+)
+
+const (
+	Pause = time.Second
+)
+
+type ObjectWithStatus interface {
+	client.Object
+	GetConditions() []metav1.Condition
+}
+
+// NewK8sTest initializes a test environment on Kubernetes.
+// It requires:
+// - A running Kubernetes cluster with a local configuration bound to it.
+// - The given set CRDs installed in that cluster
+func NewK8sTest(ctx context.Context, crds ...string) (client.Client, error) {
+	kubeClient, err := TestKubeClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to setup Kubernetes test env client: %w", err)
+	}
+
+	for _, targetCRD := range crds {
+		if err := assertCRD(ctx, kubeClient, targetCRD); err != nil {
+			return nil, fmt.Errorf("failed to asert for test-required CRD: %w", err)
+		}
+	}
+	return kubeClient, nil
+}
+
+// TestKubeClient returns a Kubernetes client for tests.
+// It requires a running Kubernetes cluster and a local configuration to it.
+// It supports core Kubernetes types, production and experimental CRDs.
+func TestKubeClient() (client.Client, error) {
+	testScheme, err := getTestScheme(
+		corev1.AddToScheme,
+		apiextensionsv1.AddToScheme,
+		akov2.AddToScheme,
+		akov2next.AddToScheme)
+	if err != nil {
+		return nil, fmt.Errorf("failed to setup Kubernetes test env scheme: %w", err)
+	}
+	return getKubeClient(testScheme)
+}
+
+func Apply(ctx context.Context, kubeClient client.Client, objs ...client.Object) error {
+	for i, obj := range objs {
+		if err := apply(ctx, kubeClient, obj); err != nil {
+			return fmt.Errorf("failed to apply object %d: %w", (i + 1), err)
+		}
+	}
+	return nil
+}
+
+func apply(ctx context.Context, kubeClient client.Client, obj client.Object) error {
+	key := client.ObjectKeyFromObject(obj)
+	old := obj.DeepCopyObject().(client.Object)
+	err := kubeClient.Get(ctx, key, old)
+	switch {
+	case err == nil:
+		obj = obj.DeepCopyObject().(client.Object)
+		obj.SetResourceVersion(old.GetResourceVersion())
+		if err := kubeClient.Update(ctx, obj); err != nil {
+			return fmt.Errorf("failed to update %s: %w", key, err)
+		}
+	case apierrors.IsNotFound(err):
+		if err := kubeClient.Create(ctx, obj); err != nil {
+			return fmt.Errorf("failed to create %s: %w", key, err)
+		}
+	default:
+		return fmt.Errorf("failed to apply %s: %w", key, err)
+	}
+	return nil
+}
+
+func CreateNamespace(ctx context.Context, kubeClient client.Client, namespace string) error {
+	return kubeClient.Create(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}})
+}
+
+func WipeNamespace(ctx context.Context, kubeClient client.Client, namespace string) error {
+	return kubeClient.Delete(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}})
```

Comment:
inline this function, it provides no value other than the reader must introspect a different package.